account_invoice_constraint_chronology/model/account_invoice.py

```diff
@@ -11,42 +11,36 @@ class AccountInvoice(models.Model):
 
     @api.multi
     def action_move_create(self):
-        previously_validated = self.filtered(lambda inv: inv.move_name)
+        previously_validated = self.filtered(lambda x: x.move_name)
         res = super(AccountInvoice, self).action_move_create()
         for inv in self:
             if inv.journal_id.check_chronology:
-                invoices = \
-                    self.search([('state', 'not in',
-                                  ['open', 'paid', 'cancel', 'proforma',
-                                   'proforma2']),
-                                 ('date_invoice', '!=', False),
-                                 ('date_invoice', '<', inv.date_invoice),
-                                 ('journal_id', '=', inv.journal_id.id)],
-                                limit=1)
+                date_field = 'date' if inv.type in ['in_invoice', 'in_refund']\
+                    else 'date_invoice'
+                date_field_value = inv.date if inv.type in ['in_invoice', 'in_refund'] \
+                    else inv.date_invoice
+                invoices = self.search([
+                    ('state', 'not in', ['open', 'paid', 'cancel', 'proforma',
+                     'proforma2']),
+                    (date_field, '!=', False),
+                    (date_field, '<', date_field_value),
+                    ('journal_id', '=', inv.journal_id.id)
+                ], limit=1)
```

Comment:
Non sarebbe il caso di ordinarle per data in modo che date_field_value contenga l'ultima utile?

Reply:
This search is only used to give an error if a record is found, no need to get a particular invoice (`invoices` variable is not used after the search)